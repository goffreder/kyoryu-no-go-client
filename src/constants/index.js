export const ui = {
    GRID_SIZE: 40,
};

export const board = {
    EMPTY: 0,
    BLACK: 1,
    WHITE: 2,
};

export const msg = {
    ATARI: 'ATARI!',
    SUICIDE: 'SUICIDE!',
    GAME_OVER: 'GAME OVER!',
};
