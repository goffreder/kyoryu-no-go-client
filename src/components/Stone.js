import React, { Component } from 'react';
import { oneOf } from 'prop-types';
import classNames from 'classnames';
import Dimensions from 'react-dimensions';

import { board as constants } from '../constants';

class Stone extends Component {
    static propTypes = {
        color: oneOf([constants.BLACK, constants.WHITE]).isRequired,
    };

    render() {
        const w = this.props.containerWidth;
        return (
            <div
                style={{
                    background:
                        this.props.color === constants.BLACK
                            ? '#101015'
                            : '#EEEEF0',
                    borderRadius: 9001,
                    boxShadow:
                        (this.props.color === constants.BLACK
                            ? 'inset 0 ' +
                              w / 6 +
                              'px ' +
                              w / 3 +
                              'px 0 rgba(255,255,255,.2),'
                            : 'inset 0 ' +
                              -w / 6 +
                              'px ' +
                              w / 3 +
                              'px 0 rgba(0,0,0,.16),') +
                        '0 ' +
                        w / 12 +
                        'px ' +
                        w / 6 +
                        'px 0 rgba(0,0,0,.16),' +
                        '0 ' +
                        w / 12 +
                        'px ' +
                        w / 4 +
                        'px 0 rgba(0,0,0,.12)',
                    position: 'absolute',
                    top: '5%',
                    right: '5%',
                    bottom: '5%',
                    left: '5%',
                }}
            />
        );
    }
}

export default Dimensions()(Stone);
