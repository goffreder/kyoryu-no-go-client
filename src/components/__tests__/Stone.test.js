import React from 'react';
import renderer from 'react-test-renderer';

import Stone from '../Stone';
import { board } from '../../constants';

describe('Stone component', () => {
    it('displays a white stone', () => {
        expect(
            renderer.create(<Stone color={board.WHITE} />).toJSON(),
        ).toMatchSnapshot();
    });

    it('displays a black stone', () => {
        expect(
            renderer.create(<Stone color={board.BLACK} />).toJSON(),
        ).toMatchSnapshot();
    });

    it('displays a white stone (last move)', () => {
        expect(
            renderer.create(<Stone color={board.WHITE} isLastMove />).toJSON(),
        ).toMatchSnapshot();
    });

    it('displays a black stone (last move)', () => {
        expect(
            renderer.create(<Stone color={board.BLACK} isLastMove />).toJSON(),
        ).toMatchSnapshot();
    });
});
