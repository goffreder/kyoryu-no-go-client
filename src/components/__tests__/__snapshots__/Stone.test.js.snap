// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stone component displays a black stone (last move) 1`] = `
<div
  className="stone stone-black"
  style={
    Object {
      "background": "#101015",
      "borderRadius": 9001,
      "bottom": "5%",
      "boxShadow": Array [
        "inset 0 3.3333333333333335px 6.666666666666667px 0 rgba(255, 255, 255, .2)",
        "0 1.6666666666666667px 3.3333333333333335px 0 rgba(0, 0, 0, .16)",
        "0 1.6666666666666667px 5px 0 rgba(0, 0, 0, .12)",
      ],
      "left": "5%",
      "position": "absolute",
      "right": "5%",
      "top": "5%",
    }
  }
>
  <div
    style={
      Object {
        "border": "0.2em solid #EEEEF0",
        "borderRadius": "100%",
        "height": "40%",
        "left": "30%",
        "position": "relative",
        "top": "30%",
        "width": "40%",
      }
    }
  />
</div>
`;

exports[`Stone component displays a black stone 1`] = `
<div
  className="stone stone-black"
  style={
    Object {
      "background": "#101015",
      "borderRadius": 9001,
      "bottom": "5%",
      "boxShadow": Array [
        "inset 0 3.3333333333333335px 6.666666666666667px 0 rgba(255, 255, 255, .2)",
        "0 1.6666666666666667px 3.3333333333333335px 0 rgba(0, 0, 0, .16)",
        "0 1.6666666666666667px 5px 0 rgba(0, 0, 0, .12)",
      ],
      "left": "5%",
      "position": "absolute",
      "right": "5%",
      "top": "5%",
    }
  }
/>
`;

exports[`Stone component displays a white stone (last move) 1`] = `
<div
  className="stone stone-white"
  style={
    Object {
      "background": "#EEEEF0",
      "borderRadius": 9001,
      "bottom": "5%",
      "boxShadow": Array [
        "inset 0 -3.3333333333333335px 6.666666666666667px 0 rgba(0, 0, 0, .16)",
        "0 1.6666666666666667px 3.3333333333333335px 0 rgba(0, 0, 0, .16)",
        "0 1.6666666666666667px 5px 0 rgba(0, 0, 0, .12)",
      ],
      "left": "5%",
      "position": "absolute",
      "right": "5%",
      "top": "5%",
    }
  }
>
  <div
    style={
      Object {
        "border": "0.2em solid #101015",
        "borderRadius": "100%",
        "height": "40%",
        "left": "30%",
        "position": "relative",
        "top": "30%",
        "width": "40%",
      }
    }
  />
</div>
`;

exports[`Stone component displays a white stone 1`] = `
<div
  className="stone stone-white"
  style={
    Object {
      "background": "#EEEEF0",
      "borderRadius": 9001,
      "bottom": "5%",
      "boxShadow": Array [
        "inset 0 -3.3333333333333335px 6.666666666666667px 0 rgba(0, 0, 0, .16)",
        "0 1.6666666666666667px 3.3333333333333335px 0 rgba(0, 0, 0, .16)",
        "0 1.6666666666666667px 5px 0 rgba(0, 0, 0, .12)",
      ],
      "left": "5%",
      "position": "absolute",
      "right": "5%",
      "top": "5%",
    }
  }
/>
`;
